torch/distributed/tensor/placement_types.py

```diff
@@ -192,8 +175,10 @@ def _shard_tensor(
         )
 
         # Only unpad if the local_tensor was padded on the dimension.
-        if pad_sizes and pad_sizes[mesh_dim_local_rank] > 0:
```

Comment:
We don't have to check for non-emptyness anymore thanks to the change in calling conventions.